mutate/mutate.go

```diff
@@ -298,6 +302,29 @@ func (m *Mutator) Add(ctx context.Context, mediaType string, r io.Reader, histor
 	return desc, nil
 }
 
+// AddExisting adds a blob that already exists to the layer, using the user
+// specified DiffID. It currently checks that the layer exists, but does not
+// validate the DiffID.
+func (m *Mutator) AddExisting(ctx context.Context, desc ispec.Descriptor, history *ispec.History, diffID digest.Digest) error {
+	if err := m.cache(ctx); err != nil {
+		return errors.Wrap(err, "getting cache failed")
+	}
+
+	// make sure the blob exists
+	present, err := m.engine.StatBlob(ctx, desc.Digest)
+	if err != nil {
+		return err
+	}
+	if !present {
+		return errors.Errorf("blob %s not present", desc.Digest)
+	}
```

Comment:
What if we didn't check for the blob's existence? It isn't really necessary for safety reasons, and if the caller really wants to make sure the blob exists they can call StatBlob (or just try to do a walk of the descriptor tree).

Reply:
Sure, I'll drop it.